src/vsr/replica.zig

```diff
@@ -4138,12 +4188,9 @@ pub fn ReplicaType(
                 return false;
             }
 
-            if (header.op <= self.commit_min) {
-                if (self.journal.header_with_op(header.op)) |existing| {
-                    // If we already committed this op, the repair must be the identical message.
-                    assert(header.checksum == existing.checksum);
-                }
-            }
+            // If we already committed this op, the repair must be the identical message.
+            assert(header.op <= self.op_checkpoint() or header.op > self.commit_min or
+                self.journal.has(header));
```

Comment:
A bit hard to untangle assert. I'd maybe preserve an if here? 

```zig
if (header.op <= self.commit_min and header.op > self.op_checkpoint()) {
    assert(self.journal.has(header));
}
```

reeealy miss the implication operator here!

Reply:
> Fwiw, it is actually

I _think_ I wrote the same thing, but skipped my own `lo < foo and foo < hi` rule, making it less readable.